File: library/src/main/java/name/zeno/android/anim/InterpolationObservableAnimation.java
package name.zeno.android.anim;

import android.view.animation.Animation;
import android.view.animation.Transformation;

/**
 * 允许监听动画时间插值
 * <p>
 * Create Date: 15/10/22
 *
 * @author 陈治谋 (513500085@qq.com)
 */
@SuppressWarnings("unused")
public class InterpolationObservableAnimation extends Animation
{
  public interface OnInterpolatedListener
  {
    void onInterpolated(float interpolatedTime);
  }

  private OnInterpolatedListener onInterpolatedListener;

  public OnInterpolatedListener getOnInterpolatedListener()
  {
    return onInterpolatedListener;
  }

  public void setOnInterpolatedListener(OnInterpolatedListener onInterpolatedListener)
  {
    this.onInterpolatedListener = onInterpolatedListener;
  }

  @Override
  protected void applyTransformation(float interpolatedTime, Transformation t)
  {
    super.applyTransformation(interpolatedTime, t);
    if (onInterpolatedListener != null) {
      onInterpolatedListener.onInterpolated(interpolatedTime);
    }
  }
}

File: library/src/main/java/name/zeno/android/anim/InterpolationObservableAnimation.kt
package name.zeno.android.anim

import android.view.animation.Animation
import android.view.animation.Transformation

/**
 * 允许监听动画时间插值
 *
 *
 * Create Date: 15/10/22
 *
 * @author 陈治谋 (513500085@qq.com)
 */
class InterpolationObservableAnimation : Animation() {

  var onInterpolatedListener: OnInterpolatedListener? = null

  interface OnInterpolatedListener {
    fun onInterpolated(interpolatedTime: Float)
  }

  override fun applyTransformation(interpolatedTime: Float, t: Transformation) {
    super.applyTransformation(interpolatedTime, t)
    if (onInterpolatedListener != null) {
      onInterpolatedListener!!.onInterpolated(interpolatedTime)
    }
  }
}
